bedrock/src/smart_account/nonce.rs

```diff
@@ -0,0 +1,170 @@
+//! A helper to construct 4337 UserOperation nonces.
+//!
+//! Layout (24-byte nonceKey + 8-byte sequence):
+//! [0]        : typeId (1 byte)
```

Comment:
This schema needs to be adjusted to consider PBH transactions. The way that PBH transactions are specified are by setting the [first 2 bytes](https://github.com/worldcoin/world-chain/blob/main/contracts/src/PBH4337Module.sol#L89). We can reduce this to 1 byte or maybe half a byte if really needed

I would propose reducing the size of magic to 4 bytes instead